### DISCUSSION TITLE
Add Spyder as a supported editor

### DISCUSSION THREAD

**Discussion by BastianZim** [community/community#7223] [Copilot Conversations]
Hi, first of all: This absolutely awesome, I'm blown away, seriously!

This is a feature request to add the[ Spyder IDE](https://www.spyder-ide.org) as a supported editor. The Spyder IDE is used in the python environment and primarily for scientific research and development. Thus, lots of frameworks and modules are used which would be a good use-case for section 6. of the [getting started](https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/gettingstarted.md#6-using-a-framework). 
It is also used by many universities and therefore by students that do not know how to code well, who would be supported by this feature.

Additionally, the whole IDE is open-source so the integration can be furthered by the community.

**Comment by pavanbhatks** [comment]
Any progress here regarding integrating Copilot for spyder

**Comment by BastianZim** [comment]
Hi @queenofcorgis and @Akash1134 apologies for pinging you out of the blue. I was just wondering if someone from GitHub could take a look at this issue? It's been open for two years and is the most upvoted unanswered one currently (I think) so there is some interest from the community. 

Maybe also cc @ccordoba12 for visibility

**Comment by ccordoba12** [comment]
Hey @BastianZim, thanks for the ping! You'll be happy to know that we requested and got approved funding from NumFocus (in the form of a Small Development Grant) to develop a code completion client for AI-based services, such as Copilot or Tabnine. We haven't decided which service to support yet, but it's really cool to know that there's a lot of interest in Copilot (judging by people's reactions to your OP).

CCing @dalthviz about this too, who's going to be in charge of that project.

**Reply by BastianZim** [reply]
Hi @ccordoba12 that's awesome news, congrats! Looking forward to that! Maybe that also convinces GitHub to pay more attention to this discussion... :)

**Reply by dalthviz** [reply]
Hi!, Just in case, the detailed proposal that we will be working on is here: https://github.com/spyder-ide/spyder/wiki/NumFOCUS-Small-Development-Grant-Spring-2023 :)

**Reply by BastianZim** [reply]
Hi everyone! Not sure how far you are with the project, but I just saw that OpenAI released an Assistants API that can be used for exactly that: https://platform.openai.com/docs/assistants/overview

As GitHub Copilot (At least in my testing) seems to be based on a more simple version, this might be an even better alternative if they keep ignoring the issue.

**Reply by dalthviz** [reply]
Hi again and thank you for the info! We will be giving that API a check :+1:

**Comment by costantinoai** [comment]
While native Copilot support for Spyder isn't available yet, I've built an open-source plugin that brings similar functionality using local LLMs: [spyder-ai-assistant](https://github.com/costantinoai/spyder-ai-assistant).

It adds Copilot-style inline completions (ghost text, Tab to accept), a dockable chat panel, and editor context awareness — all running locally via Ollama. No API keys, fully offline.

Cloud providers (OpenAI, Anthropic) are on the roadmap. It's in beta but functional. Feedback welcome!
